server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/SetMessagesMethodStepdefs.java

```diff
@@ -99,40 +115,111 @@ public void copyMessageToMailbox(String message, String sourceMailbox, String de
+        userStepdefs.execWithUser(username, () -> copyMessageToMailbox(message, sourceMailbox, sourceUser, destinationMailbox, destinationUser));
+    }
+
+    @When("^the user copy \"([^\"]*)\" from mailbox \"([^\"]*)\" of user \"([^\"]*)\" to mailbox \"([^\"]*)\" of user \"([^\"]*)\"")
+    public void copyMessageToMailbox(String message, String sourceMailbox, String sourceUser, String destinationMailbox, String destinationUser) throws Throwable {
+        MessageId messageId = getMessagesMethodStepdefs.getMessageId(message);
+        MailboxId sourceMailboxId = mainStepdefs.jmapServer
+            .getProbe(MailboxProbeImpl.class)
+            .getMailbox(MailboxConstants.USER_NAMESPACE, sourceUser, sourceMailbox)
+            .getMailboxId();
+        MailboxId destinationMailboxId = mainStepdefs.jmapServer
+            .getProbe(MailboxProbeImpl.class)
+            .getMailbox(MailboxConstants.USER_NAMESPACE, destinationUser, destinationMailbox)
+            .getMailboxId();
+
+        post("[" +
+            "  [" +
+            "    \"setMessages\","+
+            "    {" +
+            "      \"update\": { \"" + messageId.serialize() + "\" : {" +
+            "        \"mailboxIds\": [\"" + destinationMailboxId.serialize() + "\",\"" + sourceMailboxId.serialize() + "\"]" +
+            "      }}" +
+            "    }," +
+            "    \"#0\"" +
+            "  ]" +
+            "]");
+        mainStepdefs.awaitMethod.run();
+    }
+
+    @Given("^\"([^\"]*)\" move \"([^\"]*)\" to mailbox \"([^\"]*)\" of user \"([^\"]*)\"")
+    public void moveMessageToMailbox(String username, String message, String destinationMailbox, String destinationUser) throws Throwable {
+        userStepdefs.execWithUser(username, () -> moveMessageToMailbox(message, destinationMailbox, destinationUser));
+    }
+
+    @Given("^the user move \"([^\"]*)\" to mailbox \"([^\"]*)\" of user \"([^\"]*)\"")
+    public void moveMessageToMailbox(String message, String destinationMailbox, String destinationUser) throws Throwable {
+        MessageId messageId = getMessagesMethodStepdefs.getMessageId(message);
+        MailboxId destinationMailboxId = mainStepdefs.jmapServer
+            .getProbe(MailboxProbeImpl.class)
+            .getMailbox(MailboxConstants.USER_NAMESPACE, destinationUser, destinationMailbox)
+            .getMailboxId();
+
+        post("[" +
+            "  [" +
+            "    \"setMessages\","+
+            "    {" +
+            "      \"update\": { \"" + messageId.serialize() + "\" : {" +
+            "        \"mailboxIds\": [\"" + destinationMailboxId.serialize() + "\"]" +
+            "      }}" +
+            "    }," +
+            "    \"#0\"" +
+            "  ]" +
+            "]");
+        mainStepdefs.awaitMethod.run();
+    }
+
+    @When("^\"([^\"]*)\" set flags on \"([^\"]*)\" to \"([^\"]*)\"")
```

Comment:
set flags a,b,c on message ... (and it should do that in Add mode)

Reply:
Sorry, but that is done at the JMAP layer, thus *Add mode* is a non sens.

I will improve the Jerkin string as you suggested.

Reply:
Ok, just think about it : if I have a mail with a label A, as a user, when I say : i set label B to my email, I don't intend to remove label A, right ? It's not a technical thing, it's plain english.
So if Addition is not possible right now we should choose other words.
It could be "robert defines message M flags as A,B,C". I think it makes it clear it "overwrite" flags.

Reply:
I just added tests about Keywords addition/removal